jena-arq/src/main/java/org/apache/jena/sparql/engine/http/HttpQuery.java

```diff
@@ -298,6 +292,19 @@ public InputStream exec() throws QueryExceptionHTTP {
             throw jEx;
         }
     }
+    
+    private void contextualizeCompressionSettings() {
+        final RequestConfig.Builder builder = RequestConfig.copy(context.getRequestConfig());
+        builder.setContentCompressionEnabled(allowGZip || allowDeflate);
```

Comment:
It's a very fair point. I was trying to be very light in my changes to the API, but this is confusing. I'll fix it by going to a single `allowCompression` setting.